pkg/reconcilers/deploymentconfig.go

```diff
@@ -13,71 +13,42 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
-func DeploymentConfigResourcesMutator(existingObj, desiredObj common.KubernetesObject) (bool, error) {
-	existing, ok := existingObj.(*appsv1.DeploymentConfig)
-	if !ok {
-		return false, fmt.Errorf("%T is not a *appsv1.DeploymentConfig", existingObj)
+// DCMutateFn is a function which mutates the existing DeploymentConfig into it's desired state.
+type DCMutateFn func(desired, existing *appsv1.DeploymentConfig) bool
+
+func DeploymentConfigMutator(opts ...DCMutateFn) MutateFn {
```

Comment:
I like this approach :+1:, makes easier adding different mutators in a single mutator